clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesJobProvider.groovy

```diff
@@ -19,27 +19,42 @@ package com.netflix.spinnaker.clouddriver.kubernetes.v1.provider.view
 import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesNamedAccountCredentials
 import com.netflix.spinnaker.clouddriver.kubernetes.v1.model.KubernetesJobStatus
 import com.netflix.spinnaker.clouddriver.kubernetes.v1.security.KubernetesV1Credentials
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.agent.KubernetesCacheDataConverter
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.view.model.KubernetesV2Manifest
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesSelectorList
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials
 import com.netflix.spinnaker.clouddriver.model.JobProvider
 import com.netflix.spinnaker.clouddriver.model.JobState
+import com.netflix.spinnaker.clouddriver.model.Manifest
+import com.netflix.spinnaker.clouddriver.model.ManifestProvider
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import io.kubernetes.client.models.V1DeleteOptions
+import io.kubernetes.client.models.V1Job
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
+import java.util.stream.Collectors
+
 @Component
 class KubernetesJobProvider implements JobProvider<KubernetesJobStatus> {
   String platform = "kubernetes"
 
   @Autowired
   AccountCredentialsProvider accountCredentialsProvider
 
+  @Autowired
+  List<ManifestProvider> manifestProviders
+
   KubernetesJobProvider() { }
 
   @Override
   KubernetesJobStatus collectJob(String account, String location, String id) {
     def credentials = accountCredentialsProvider.getCredentials(account)
-    if (!(credentials?.credentials instanceof KubernetesV1Credentials)) {
-      return null
+    if (credentials?.credentials instanceof KubernetesV2Credentials) {
```

Comment:
Rather than doing an `if/else` on the type of the `credentials` it probably makes sense to just make a new `KubernetesV2JobProvider` that implements the `JobProvider<KubernetesJobStatus>` interface.  It looks like there's actually no shared logic between the V1/V2 implementations, so having them in different classes will make it more maintainable.